src/entry.rs

```diff
@@ -631,9 +635,11 @@ impl<'a> EntryFields<'a> {
                 })?;
             }
         }
+        #[cfg(not(target_os = "wasi"))]
```

Comment:
I think this would make more sense as a conditional definition of `set_perms` for wasi (which I think already exists otherwise for windows/unix)

Reply:
This will now return an error if someone uses this library with WASI and turns the preserve_permissions flag on.
Note if someone changes the default in the future this will now cause many builds to fail but changing the default on this flag would be a breaking change anyway so this is kind of already a risk